client/azure_client_set.go

```diff
@@ -82,20 +44,21 @@ type AzureClientSet struct {
 
 // NewAzureClientSet returns the Azure API clients.
 func NewAzureClientSet(config AzureClientSetConfig) (*AzureClientSet, error) {
-	if err := config.Validate(); err != nil {
-		return nil, microerror.Maskf(invalidConfigError, "config.%s", err)
+	err := config.Validate()
+	if err != nil {
+		return nil, microerror.Mask(err)
 	}
 
 	// Returns environment object contains all API endpoints for specific Azure
 	// cloud. For empty config.EnvironmentName returns Azure public cloud.
 	env, err := parseAzureEnvironment(config.EnvironmentName)
 	if err != nil {
-		return nil, microerror.Maskf(err, "parsing Azure environment")
+		return nil, microerror.Mask(err)
 	}
 
 	servicePrincipalToken, err := newServicePrincipalToken(config, env)
 	if err != nil {
-		return nil, microerror.Maskf(err, "creating service principal token")
```

Comment:
I find these super useless to be honest. It does not add value during debugging.

Reply:
As long as the original error has enough context for debugging, I agree.

Reply:
The stack is the context. At least this is for me the most relevant thing. The error message almost never tells me anything useful. I look at the stack. I look at the code. I fix the problem.